tests/cli-self-upd.rs

```diff
@@ -330,8 +330,8 @@ fn install_adds_path_to_profile() {
 
 #[test]
 #[cfg(unix)]
-fn install_adds_path_to_bash_profile() {
-    install_adds_path_to_rc(".bash_profile");
+fn install_adds_path_to_bashrc() {
```

Comment:
Won't this lead to ~/.cargo/bin being in the path as many times as subshells have been entered? Thats why ~/.profile is used for this kind of thing usually - that or more complicated once-and-only-once constructs.

Secondly, adding to bashrc has the negative effect of *not* adding to PATH when someone ssh's into their environment and runs rustup non-interactively.

I think both of these are addressable. But perhaps I'm wrong and your patch does - but there's no documentation update explaining cover message explaining the overall change so I'm a bit unclear on the big picture and just inferring it from the commit contents :).

The google cloud sdk does the sort of idempotent shell shenanigans I'm talking about; adding that into rc files in addition to keeping the profile additions would suffice I think.

Reply:
So, here's the key problem: `.bash_profile` matters not-at-all for the 90% case of a GUI terminal user.
The primary way this is addressed lately is by having .bash_profile `source ~/.bashrc`.
Modifying the exact shell code so it doesn't add it twice is simple, but rustup currently accomplishes this using .bash_profile and thereby makes it so that for most developers, it appears 0 times in PATH.

Pushing it into both .bash_profile and .bashrc could be done but seems like succumbing to madness (since it results in duplication, and thus it is more opaque to the user how to remove it if they need to manually edit it).

Reply:
The Google Cloud SDK is actually in fact also a wonderful example of "this doesn't actually work for many users precisely because it updates .bash_profile over .bashrc when .bash_profile isn't loaded when I start a terminal."

Reply:
google cloud SDK updated rc files when I installed it recently for my new job. Perhaps I did something unusual there.

I appreciate the issues for GUi terminal users, and am sympathetic to them. I use Linux all day every day, but I don't use a Linux GUI (well, I do use i3 from time to time, but thats a digression). 

I'd like to make sure we meet everyones needs.

Reply:
NB: Discussion elaborated and the conclusion arrived at is that actually, descending into a little madness is fine, but we still have to use an idempotent framework for modifying PATH, and this is unavoidable because all of this might have made sense at one time, but now it's 2020. Further explanation will be documented for future reference in followup commit.